external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala

```diff
@@ -67,6 +67,13 @@ private[sql] class AvroOptions(
     })
   }
 
+  /**
+   * Iff true, perform Catalyst-to-Avro schema matching based on field position instead of field
+   * name. This allows for a structurally equivalent Catalyst schema to be used with an Avro schema
+   * whose field names do not match. Defaults to false.
+   */
+  val schemaRenaming: Boolean = parameters.get("schemaRenaming").exists(_.toBoolean)
```

Comment:
shall we make it "writeByPosition"

Reply:
This is used on the read side as well, so I don't think "writeByPosition" is correct. How about "positionalFieldMatching" or "matchFieldsByPosition"?

Reply:
Updated to `positionalFieldMatching` in the latest diff.